packages/nextjs/hooks/useHeimdall.ts

```diff
@@ -0,0 +1,47 @@
+import { useQuery } from "@tanstack/react-query";
+import { Address, isAddress } from "viem";
+import { HEIMDALL_API_URL } from "~~/utils/constants";
+
+type UseHeimdallParams = {
+  contractAddress?: Address;
+  rpcUrl?: string;
+  disabled?: boolean;
+};
+
+export const useHeimdall = ({ contractAddress, rpcUrl, disabled = false }: UseHeimdallParams) => {
+  const fetchFromHeimdall = async () => {
+    if (!contractAddress || !isAddress(contractAddress)) {
+      throw new Error("Invalid contract address");
+    }
+    if (!rpcUrl) {
+      throw new Error("RPC URL is required");
+    }
+
+    const rpcUrlWithoutHttps = rpcUrl.substring(8);
+    const response = await fetch(`${HEIMDALL_API_URL}/${contractAddress}?rpc_url=${rpcUrlWithoutHttps}`);
```

Comment:
Nitpick: Maybe we also add a check `if(!response.ok) throw new Error("Interal server error")` or something. 

because `fetch` doesn't throw error when responds with 404 or other any other status code : https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch